kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncThreadPool.java

```diff
@@ -33,6 +33,11 @@ public class AsyncThreadPool {
       = new HashMap<>();
   private final BlockingQueue<Runnable> processingQueue;
 
+  // TODO: we don't really need to track this by processor/partition, technically an
+  //  error anywhere is fatal for the StreamThread and all processors + all further
+  //  processing should shut down ASAP to minimize ALOS overcounting
```

Comment:
This is what I mentioned on slack: we should really consider an exception on any partition to be fatal for all, and furthermore actually suspend the AsyncThreadPool altogether (ie cancel all in-flight events) as soon as we see a fatal error anywhere. But I'll leave this for followup